### ISSUE TITLE
BUG - 3 - All - Accessibility: Pagination Date Range selection issues

### ISSUE THREAD

**Issue by SKMOstudio** [department-of-veterans-affairs/va-mobile-app#11665]
## What happened?

When a screen reader user is on a page that contains the Date Range picker (Health > Travel and Past Appointments), the screen reader has some inconsistent behaviors:

iOS:
1. Focus on drop-down range menu doesn't read out selected option.
2. When reading the months, some of them are not pronounced correctly. (June - "Joon", January - "Jan", March - "Mar", December - "Deck")

Android:
3. The Date Picker object on the page has multiple "focuses" (date picker, date picker + results text, date picker + results text + list of content) that the user has to swipe through to navigate to other content on the page.

## Specs:

- Device: iOS, Android
- OS: N/A
- User Account: DEMO
- Accessibility State: Screen reader ON, default settings.

## Steps to Reproduce

1.1 - On an iOS device, turn the screen reader on.
1.2 - Navigate to the Health > Travel page.
1.3 - Navigate down the page until the date range picker is in focus. Observe - The screen reader does not mention the current selection in the picker.

2.1  - On an iOS device, open the date range menu.
2.2 - Scroll down in the menu to a date range that includes months. (ex: Oct 2024 - Dec 2024).
2.3 - Observe: The screen reader does not read out the abbreviation of the months out properly.

3.1 - On an Android device, turn the screen reader on.
3.2 - Navigate to the Health > Travel page.
3.2 - Navigate through the page elements until the date range picker is in focus. Observe: You should see the page 'highlight' the date ranger picker multiple times, each one encompassing a smaller subset of other page elements.

## Desired behavior

1. When a screen reader user focuses on the date picker object, the screen reader should read out the menu object, the current selection made, and the current number of results for the chosen object (ex: "Showing 1 - 10 of 12 results").
2. When a screen reader user focuses on the date range within the Date Picker menu, the screen reader should read out the abbreviated months ("Jan", "Mar", "Dec") as if it was the full name of that month.
3. When a user focuses on the Date Picker object, it should only focus on a single selectable element.

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: High **Low** 
- Frequency: **High** Low 

- 3 - **Low**
- 2 - High
- 1 - Critical

## Screen shot(s) and additional information

https://github.com/user-attachments/assets/3eda3a14-bc7b-48dc-9f7e-9ed2d832124e
https://github.com/user-attachments/assets/2fe6bb3b-bd7a-41e1-89e7-533c457b0e70

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, global, Health, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the relevant team's bug epic ([Global](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8302), [Health & Benefits](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8303), [API](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8304), [QART](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8305))
